src/button.tsx

```diff
@@ -291,7 +292,18 @@ const Button = React.forwardRef<TouchableElement, ButtonProps & {type: ButtonTyp
     }
 
     if (props.onPress) {
-        return <BaseTouchable {...commonProps} onPress={props.onPress} />;
+        return (
+            <BaseTouchable
+                {...commonProps}
+                onPress={(e) => {
+                    const result = props.onPress(e);
+                    if (result) {
```

Comment:
if (result?.then)

You don't know at this point if the onPress is returning undefined, null, a value or a promise. To check if it is a promise you can check if it has the 'then' property. I don't know if there is a better way tbh

Reply:
discussed offline, If result has a value, it will always be a promise, so this is not a promise.